salt/utils/jinja.py

```diff
@@ -196,7 +196,7 @@ def get_source(self, environment, template):
             }
             environment.globals.update(tpldata)
 
-        if _template in self.cached:
+        if _template in self.cached or os.path.exists(_template):
```

Comment:
I think the proper check needs to be something like `if _template in self.cached or (os.path.exists(_template) and os.path.isabs(_template)):

Reply:
From https://github.com/saltstack/salt/pull/62629#discussion_r964743556 :

> "I don't think it's security-relevant whether the path is absolute or relative if we're deciding that we're allowing access outside the salt filesystem (retaining pre-existing functionality)."

While the title of the PR contains "absolute path" because that's what was reported, the `exists` logic reverts the outcome to the same that was available prior to the patch in 3005.

Reply:
Let's retain the previous functionality.